Python bindings

ftalbrecht opened the pull request:
This introduces python bindings, see https://github.com/dune-community/dune-xt-common/pull/24 and https://github.com/dune-community/dune-xt-grid/pull/14. Things we might want to change (though not necessarily in this pr):
- Add a good way to (not manually) add functions for all grids/dimension, etc...

**Before we merge** we also need to revert e5d79201cfabeef546739c05fa3eb560bbda4c14.

Commit:
[interfaces] add first draft for bindings

Commit:
[checkerboard] add first draft of bindings

Commit:
[constant] add first draft of bindings

Commit:
[expression] add first draft of bindings

Commit:
[spe10] add first draft of bindings for model1 function

Commit:
add python bindings as target `functions`

Commit:
[travis] REVERT BEFORE MERGE: use correct branch in super module

Commit:
[cmake] drop useless statement

renefritze commented on dune/xt/CMakeLists.txt:
```diff
@@ -9,4 +9,9 @@ add_subdirectory(functions)
 
+if(dune-pybindxi_FOUND)
+  dune_pybindxi_add_module(functions functions/bindings.cc)
+  target_link_dune_default_libraries(functions)
```
I think this shouldn't be necessary when using the DUNE's simple cmake mode properly in the `dune_pybindxi_add_module` macro

ftalbrecht replied:
You mean the `target_link_dune_default_libraries` part?

renefritze replied:
Yeah. That should automatically if the target was added as a `dune_add_library`

Commit:
[bindings] add type casters from dune-xt-common

Commit:
[cmake] disable bindings for gcc <= 4.9

Commit:
fix typo

Commit:
[bindings] YaspGrid -> ALUGrid

Commit:
[bindings] make use of FieldMatrix type converters

Commit:
[bindings] bind other dimensions, which are of higher interest

Commit:
[bindings] use type converter for stl containers

Commit:
[bindings] add init_logger

Commit:
[vcsetup] bump

Commit:
[travis] use generated yml

Commit:
[travis] install parmetis

Dune and the old alugrid only work together if alugrid and metis and
parmetis are installed!

Commit:
[bindings] allow more than one grid, guard alugrid

Commit:
Revert "[travis] install parmetis"

This reverts commit a74ea2d97fedd495153fcfec520bb590f38c9f25.

Commit:
[travis] use generated yml

Commit:
[pylicense] update config

Commit:
[bindings] update header

Commit:
[bindings] update infrastructure

* functions.so -> _functions.so to allow for magic in __init__.py
* create module infrastructure in build tree via symlinks during cmake

Commit:
[bindings] enable for non gcc compilers

Commit:
[interfaces] make all dims constexpr

Commit:
[bindings] work around a gcc-4.9 bug

Commit:
[cmake] create separate target for bindings

Commit:
[bindings] add more to lib if dune-pybindxi was found

Commit:
[license] add CMakeLists.txt to cfg

Commit:
start to get rid of the ChooseVariant trick, this does not work for libs

(since they get instantiated anyway)

Commit:
Revert "[bindings] add more to lib if dune-pybindxi was found"

This reverts commit d636adae63ccd04409c975f17fb7d5c4c7bdd97f.

Commit:
[license] ignore .vcsetup in cfg

Commit:
[test] add empty

Commit:
[bindings] add UGGrid<2>

ftalbrecht commented:
Done in 25d226e87fbf70d6ae3bfd4cf7ab9f5d35aa3f16.
